docs/docs/multipart.md

```diff
@@ -0,0 +1,261 @@
+which is an effect. We can call methods on the `Multiparts` companion object to get a `Multiparts[_]` instance, which is 
+a builder of multipart requests. This instance can be shared. This is an example of the creation of a multipart request:
+
+```scala mdoc:silent
+Multiparts.forSync[IO].flatMap(multiparts =>
+  multiparts.multipart(
+    Vector( // a multipart request with two parts
+      Part.fileData[IO]( // there are overloads for fileData that read directly from a file
+        name = "picture",
+        filename = "sunset.jpg",
+        entityBody = fs2.Stream.range[IO, Int](0, 100).map(_.toByte),
+        headers = `Content-Type`(MediaType.image.jpeg)
+      ),
+      Part.formData[IO](name = "description", value = "A sunset")
+    )
+  )
+)
+  .map(multipartRequest =>
+    Request[IO](
+      method = Method.POST,
+      uri = uri"http://example.com/",
+      headers = multipartRequest.headers // set the headers related to this multipart request
+    ).withEntity(multipartRequest)
+  )
+```
+
+Here's a full example with a client and a server:
+```scala mdoc:silent:nest
+import org.http4s.client.Client
+import cats.effect._
+import cats.syntax.all._
+import org.http4s._
+import org.http4s.dsl.io._
+import org.http4s.headers._
+import org.http4s.multipart._
+import org.http4s.implicits._
+
+import cats.effect.unsafe.implicits.global
+
+val routes = HttpRoutes
+  .of[IO] { case request @ POST -> Root / "multipart-form" =>
+    EntityDecoder.mixedMultipartResource[IO]().use(decoder =>
+      request.decodeWith(decoder, strict = true) { multipart =>
+        val picture = multipart.parts.find(_.name.contains("picture"))
+        val pictureSize = picture.traverse(_.body.compile.count).map(_.getOrElse(0L))
+
+        val description = multipart.parts.find(_.name.contains("description"))
+        val descriptionText = description.traverse(_.bodyText.compile.string).map(_.getOrElse(""))
+
+        (pictureSize, descriptionText)
+          .flatMapN((size, description) => 
+            Ok(s"This is a $size byte file, with the description '$description'")
+          )
+      }
+    )
+  }
+
+val client = Client.fromHttpApp(routes.orNotFound)
+val request = Multiparts.forSync[IO].flatMap(multiparts =>
+    multiparts.multipart(
+      Vector(
+        Part.fileData[IO](
+          name = "picture",
+          filename = "sunset.jpg",
+          entityBody = fs2.Stream.range[IO, Int](0, 100).map(_.toByte),
+          headers = `Content-Type`(MediaType.image.jpeg)
+        ),
+        Part.formData[IO](name = "description", value = "A sunset")
+      )
+    )
+  )
+  .map(multipartRequest =>
+    Request[IO](
+      method = Method.POST,
+      uri = uri"http://example.com/multipart-form",
+      headers = multipartRequest.headers
+    )
+      .withEntity(multipartRequest)
+  )
+
+```
+```scala mdoc
+request.flatMap(client.expect[String](_))
+  .unsafeRunSync()
+```
```

Comment:
I have not rendered these docs myself yet, is the output here interesting?
If not, perhaps this should also be a runnable scala-cli example?